### ISSUE TITLE
Develop & Design a LearnK8s Newsletter and related tasks

### ISSUE THREAD

**Issue by keithmifsud** [learnk8s/learnk8s.io#89]
As per discussion on #GH-47 and learnk8s/team#15 The landing page should match the following design:

![image](https://user-images.githubusercontent.com/2071560/41024493-215bcfd0-698f-11e8-9d19-cbdca35685d7.png)
_mockup#1_

and the newsltter itself should match this one:

![image](https://user-images.githubusercontent.com/2071560/41026066-4e5bfd76-6993-11e8-8727-aab89f48a3e9.png)
_mockup#2_

The deliverables of this EPIC are:

- [ ] A Landing Page for the newsletter, encouraging subscription as per _mockup#1_
  - [ ] Reviewed?.

- [x] A HTML template for the newsletter itself as per _mockup#2_
    - [ ] Reviewed?.

- [x] A Newsletter Archive Page containing a list of past newsletters. - may need to be revised once we have more than one or a few publications.
  - [x] Reviewed?.

- [ ] A Newsletter email template stored in _sendinblue_ containing the newsletter itself with some changes to the footer (such as a link to unsubscribe) and removal of the top navigation. __(current issue is how to get the CSS -> maybe using an external link? -> depends on _sendinblue_.__
  - [ ] Reviewed?.

- [x] A dedicated list of subscribers hosted on _sendinblue_
  - [ ]  Reviewed?.

__Please do not merge the PR #GH-90 until all the above tasks have been ticked off !!!__

## Development tasks

__Initial:__

- [x] Develop a `_newsletter` collection.
  - [x] Add permalink and output properties.
- [x] Create a blank archive page in `newsletter-archive/index.html`
  - [x] Include a draft for loop for issues within the collection so that I 
  can test links etc.

- [x] Create a blank landing page for the newsletter subscription
  - [x] Should be served under `/newsletter`.
  - [x] Include a draft link to the archive page and test permalink (always a
   headache). 

- [x] Add a menu item in top navigation of the website to the newsletter 
landing page.
  
- [x] Add a menu item in footer of the website to the newsletter landing page.

### Newsletter template:

- [x] Setup a dummy / first issue newsletter under 
`_newsletter/issue-1/index,md`.

- [x] Setup a newsletter template which will handle all front matter and includes for all the sections of the newsletter. 

- [x] Must include SEO and OG data.

- [x] Should have an OG image. I think the image should be the same across all issues.

- [x] Should include the GA snippet.

- [x] Fix issue with images not getting loaded (as tested on netflify: https://deploy-preview-90--learnk8s.netlify.com/newsletter/issue-1)

- [x] Test that once the image issue is fixed, the image is tweeted from the meme include.
  - [x] This is not possible unless the image is already on twitter and a twitter pic url is supplied. Refer to the `_pages/newsletter/README.md` file for more information.

- [x] GitHub's star and fork buttons should link to their respective functionality. Starring cannot be linked (without considerable JS), but forking can.

- [x] Ability to add content at the end of the newsletter's issue.

 - [x] Abiity to add custom content __"easily"__.
  - [x] Must be able to specify the sort order for each section, including the file's content.

- [x] Include the static "Enjoyed this issue?" section.

- [x] Include a link to the archive.

- [x] Include the static footer.

- [x] Design the newsletter view for desktop and mobile screens.
  - [x] Must replicate some styles for the includes (containing sections)

- [x] Awaiting review and action for the font-family
- [x] Awaiting decision on the skype icon.
- [x]  Awaiting instructions regarding the `dib` class on anchor tags.

### Archive Page:

- [x] Includes a reverse-chrono list of past issues.
  - [x] with the issue number and date of publication as the title.
  - [x] Includes an excerpt to summarise the issue.

- [x] Issue links should open in a separate tab because the newsletter itself does not have the website's navigation and thus the reader won't be able to come to the archive page _unless they hit the back button or use the back navigation shortcut.

- [x] Finalise design:
  - [x] Large screens
  - [x] Medium screens
  - [x] Small screens

- [x] Submit for review.
- [x] Mark as complete.

### Landing Page:

__Subscription development tasks:__

- [x] Get API (or similar) details from SendInBlue to link the subscription form with the new contact list.
  - [x] WE obviously cannot use any server-side tech in a static site! Silly me!.
- [x] Develop an initial subscribe form and link to SendInBlue.
- [x] Test Integration.
- [x] Test validation for the form fields.
- [x] Try to reduce and localise the SendInBlue dependencies .

 - [x] Setup additional tasks for the response page after subscribing - depending on how SendInBlue handles this. NO need of additional tasks other than designing the notifications.

- [x]  Review tasks for the design of a `welcome email` - depending on how SendInBlue handles this. No longer needed. NO emails will be sent out as there's a confirmation message on screen.

__Issue Preview:__

A preview of the latest issue should be automatically displayed.

- [x]  Develop an `include` that gets an iFrame (or similar tech) of the latest issue. The preview will be cropped to match the mockup.

- [x] Include the preview on the landing page.

- [x] Refactor this so that if an issue number is specified as a front-matter property, then this issue will be previewed as opposed to the latest one.

__Contributors:__

These will be manually set in front-matter until we have actual user contributions as per #95.

- [x] Develop `_includes/newsletter/contributors/static.html`.
- [x] Include in the landing page with the front-matter for 3 contributors.

__Design:__

- [x] Confirm the submissions' email address.
- [x] Add a link for user submission.
- [x] Design the landing page:
  - [x] Large screens.
  - [x] Medium screens.
  - [x] Small screens.

- [x]  Ask for review.
- [x] Complete the task.

### Newsletter email template:
- [ ] Must include an unsubscribe link.
- [ ] Must closely match the browser version of the newsletter.

### Layout

- [x] Add Google Tag Manager to newsletter layout as per #105 

### Documentation:

- [ ] Prepare and test documentation under `_pages/newsletter/README.md`
  - [ ] Should describe how to add custom sections.
- [ ] Add a link to the above from the main readme file.

### Release:

- [x] Waiting response regarding the existing contacts on Netlify and the existing subscribe form in footer.

- [x] Remove the dummy issue from `_newsletter/issue-1`.
  - [x] Not sure if possible but maybe name the dummy issue as `issue-0` and then remove it from the archive list but have it available as a preview in the landing page. Possibly add meta tags or an entry in robots file so that it is not indexed.
- [x] Update link to the archive in landing page so that it doesn't show up unless there are past issues (excl. the dummy one).

**Comment by keithmifsud** [comment]
@danielepolencic I think we should always use the same image for the open graph across all issues of the newsletter. As opposed to one per issue. Thoughts? In any case, will you supply an image? 😈

**Comment by keithmifsud** [comment]
@danielepolencic Which email address will the newsletter be sent out from? Will it also be a reply to email? You also mentioned a different address for readers to submit links for future issues. What will that email address be?

**Comment by keithmifsud** [comment]
@danielepolencic Should I import the contact from the `subscribe-blog` list on Netlify to the new newsletter subscriber's list on SendinBlue? I also think it's important to have these contacts on the same platform, so, I should link all the forms to sendinblue. Right?

**Comment by danielepolencic** [comment]
> Which email address will the newsletter be sent out from? Will it also be a reply to email? You also mentioned a different address for readers to submit links for future issues. What will that email address be?

news@learnk8s.io weekly@learnk8s.io ? yes, it will be a reply email too. Readers should be able to reply if they wish to engage or submit links.

> Should I import the contact from the subscribe-blog list on Netlify to the new newsletter subscriber's list on SendinBlue? I also think it's important to have these contacts on the same platform, so, I should link all the forms to sendinblue. Right?

Not necessary for the time being. We can do that later on. But yes, eventually those should be on the same list.

**Comment by keithmifsud** [comment]
Thank you @danielepolencic. All fine with the email address. I'll also use the same one for submitting links (i.e.: the mailto link on the landing page). Also fine for the contacts but please do keep an eye on new subscribers through Netlify because they will miss out on issues if we don't add them to the new platform.

**Comment by keithmifsud** [comment]
@danielepolencic Please setup the email address(es). They don't exist as of yet and I need one for SendInBlue. Thanks.

**Comment by keithmifsud** [comment]
@danielepolencic I'll go for news@learnk8s.io for all incoming and outgoing emails related to the newsletter. Please set the address up and let me know once done. Thanks.

**Comment by danielepolencic** [comment]
created news@ and weekly@ 👍 
You're in the group too. If you send an email to those, you should receive it.

**Comment by keithmifsud** [comment]
@danielepolencic 

I sent an email to weekly@learnk8s.io and got this:

Hello mifsud.k@gmail.com,

We're writing to let you know that the group you tried to contact (weekly) may not exist, or you may not have permission to post messages to the group. A few more details on why you weren't able to post:

 * You might have spelled or formatted the group name incorrectly.
 * The owner of the group may have removed this group.
 * You may need to join the group before receiving permission to post.
 * This group may not be open to posting.

If you have questions related to this or any other Google Group, visit the Help Center at https://support.google.com/a/learnk8s.io/bin/topic.py?topic=25838.

Thanks,

learnk8s.io admin

----

Also, when I tried to login to keith@lea... I'm asked to contact the admins to recover my password.

**Comment by danielepolencic** [comment]
Deprioritised.
